packages/core/src/tracing/spanBuffer.js

```diff
@@ -179,6 +189,14 @@ exports.addSpan = function (span) {
     return;
   }
 
+  // Process the span, apply any transformations, and implement filtering if necessary.
+  const processedSpan = processSpan(span);
+  if (!processedSpan) {
+    logger.warn('Span of type %s has no trace ID. Not transmitting this span', span.n);
```

Comment:
issue: The error message here belongs to the validation you have added. Connected to comment in line 509. If we filter out a span, we also return null. As suggested, please separate validation & processing.

Reply:
Refactored the code.